### PR TITLE
[CPDLP-2001] Improve some api error messages

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

- Ticket: [CPDLP-2001](https://dfedigital.atlassian.net/browse/CPDLP-2001)

Review the existing error message and update content to ensure the messages are useful and reflect the error. 

### Changes proposed in this pull request

Improve some api error messages

[CPDLP-2001]: https://dfedigital.atlassian.net/browse/CPDLP-2001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -55,24 +55,24 @@ en:
   missing_evidence_held: "The property '#/evidence_held' must be present"
   missing_has_passed: "The attribute '#/has_passed' must be included as part of 'completed' declaration submissions. Values can be 'true' or 'false' to indicate whether the participant has passed or failed their course"
   missing_lead_provider: "The lead provider must be present"
-  missing_course_identifier: "The property '#/course_identifier' must be present"
+  missing_course_identifier: "The  '#/course_identifier' is missing from your request. Please include a course_identifier value and try again."
```
Comment (mooktakim) [inline config/locales/en.yml:58]: There's an extra space here after "The"

File: config/locales/en.yml [hunk 2]
```diff
@@ -94,8 +94,8 @@ en:
     less_than: *default_message
     greater_than_or_equal_to: *default_message
   npq_application:
-    missing_npq_application: "The property '#/npq_application' must be present"
-    has_another_accepted_application: The participant already has an accepted application for this course
+    missing_npq_application: "The '#/npq_application' is missing from your request. Check details and try again."
```
Comment (mooktakim) [inline config/locales/en.yml:97]: I think this should be "The entered '#/npq_application'" (to stay consistent with previous changes).

**Review by mooktakim — CHANGES_REQUESTED** [review]
Couple of minor changes

**Review by cwrw — COMMENTED** [review]
looks good past the other comments, I was wondering if there were any discussions/tickets relating to having multiple errors (where one is relevant and the others aren't) in the declaration endpoint which can confuse providers, and instead surfacing only the relevant error?

**Review by mooktakim — APPROVED** [review]

**Review by cwrw — APPROVED** [review]
⭐ !

**Pull request merged**